extensions/functions_string.yaml

```diff
@@ -248,3 +251,177 @@ scalar_functions:
             name: "replacement"
             description: The replacement string.
         return: "varchar<L1>"
+  -
+    name: ltrim
+    description: >-
+      Remove any occurrence of the characters from the left side of the string.
+      If no characters are specified, spaces are removed.
+    impls:
+      - args:
+          - value: "varchar<L1>"
+            name: "input"
+            description: "The string to remove characters from."
+          - value: "varchar<L2>"
+            name: "characters"
+            description: "The set of characters to remove."
+        return: "varchar<L1>"
+      - args:
+          - value: "string"
+            name: "input"
+            description: "The string to remove characters from."
+          - value: "string"
+            name: "characters"
+            description: "The set of characters to remove."
+        return: "string"
+  -
+    name: rtrim
+    description: >-
+      Remove any occurrence of the characters from the right side of the string.
+      If no characters are specified, spaces are removed.
+    impls:
+      - args:
+          - value: "varchar<L1>"
+            name: "input"
+            description: "The string to remove characters from."
+          - value: "varchar<L2>"
+            name: "characters"
+            description: "The set of characters to remove."
+        return: "varchar<L1>"
+      - args:
+          - value: "string"
+            name: "input"
+            description: "The string to remove characters from."
+          - value: "string"
+            name: "characters"
+            description: "The set of characters to remove."
+        return: "string"
+  -
+    name: trim
+    description: >-
+      Remove any occurrence of the characters from the left and right sides of
+      the string. If no characters are specified, spaces are removed.
+    impls:
+      - args:
+          - value: "varchar<L1>"
+            name: "input"
+            description: "The string to remove characters from."
+          - value: "varchar<L2>"
+            name: "characters"
+            description: "The set of characters to remove."
+        return: "varchar<L1>"
+      - args:
+          - value: "string"
+            name: "input"
+            description: "The string to remove characters from."
+          - value: "string"
+            name: "characters"
+            description: "The set of characters to remove."
+        return: "string"
+  -
+    name: lpad
+    description: Left-pad the string with the characters until the specified length of the string has been reached.
+    impls:
+      - args:
+          - value: "varchar<L1>"
+            name: "input"
+            description: "The string to pad."
+          - value: i32
+            name: "length"
+            description: "The length of the output string."
+          - value: "varchar<L2>"
+            name: "characters"
+            description: "The set of characters to use for padding."
```

Comment:
I understand the "set of characters" concept for trim functions (`trim("foobar", "of") -> "bar"`), but not for padding. "Set" implies unordered. I don't think I've ever seen these functions in forms where you can specify more than one character either, but I guess Substrait doesn't really have a character type except maybe `fixedchar<1>`. Do these functions just repeat the padding string as many times as needed? The question then becomes what they do if the number of characters needed is uneven, especially for the `center` functions, where I can think of at least two similarly efficient algorithms that would have different results: `center("x", 10, "abc")` -> `replace_slice(substring(repeat("abc", 4), 1, 10), 5, 1, "x")` -> `"abcaxcabca"`, and `center("x", 10, "abc")` -> `concat(substring(repeat("abc", 2), 1, 4), "x", substring(repeat("abc", 2), 1, 5))` -> `"abcaxabcab"`.

Reply:
Good point.  I'll update the wording to 'string'.  DuckDB only allows one character, but postgresql let's you use a string.  I figured we would use whatever is more flexible. 

```
Do these functions just repeat the padding string as many times as needed? The question then becomes what they do if the number of characters needed is uneven
```
Yeah, if the uneven case occurs, the padding will get truncated (for postgresql): 
lpad('abc', 10, 'XYZ') -> 'XYZXYZXabc'

To be honest, I wasn't too convinced on the center function.  I couldn't really find many places that implement it.  I imagine the expectation is that someone could easily use a combination of lpad and rpad.  I wonder if we should omit here as well?  @iancook

Reply:
Another thought is that maybe we have center only take single character padding.  Multiple character strings used for padding can be just for lpad/rpad

Reply:
@jvanstraten in the interest of getting a bunch of these other more common functions in, i've removed center for now.  I'll create a separate task to look more into center and if/how other SQL variants deal with it.